### PR TITLE
Added check for running dev server

### LINKED ISSUE #1343: `CI/CD` : Check for Starting a Server.
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is.

For PR workflow under talawa-api repository, there should be a job that checks whether the development server starts successfully or not.

**Describe the solution you'd like**

A job that checks whether the apollo server can start or not.

**Describe alternatives you've considered**

**Approach to be followed (optional)**
A clear and concise description of approach to be followed.

**Additional context**

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by aarishshahmohsin**
**What kind of change does this PR introduce?**

Added Check for Starting a Server.

**Issue Number:**

Fixes #1343 

**Did you add tests for your changes?**
No

**Snapshots/Videos:**

**If relevant, did you update the documentation?**

**Summary**
This change will add a check for Starting a Server.

**Does this PR introduce a breaking change?**

No
**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

**Comment by palisadoes** [comment]
@xoldyckk We are adding this test to make sure the app starts as part of the approval process. We have had cases where the PR tests pass, but the app doesn't start. What are your thoughts?

**Comment by palisadoes** [comment]
OK, let's do that.

**Comment by palisadoes** [comment]
It sounds like you should create a health check endpoint that returns data similar to what was suggested and then query it as part of the testing regime.

The data would need to return `healthy` if underlying basic checks are OK

**Comment by aarishshahmohsin** [comment]
```yaml
  - name: GraphQL Health Check
      run: |
        healthCheckResponse=$(curl -X POST -H "Content-Type: application/json" \
          -d '{ "query": "{ HealthCheck { healthCheck { status } } }" }' \
          http://localhost:4000/graphql)
        
        if [[ $healthCheckResponse == *"healthy"* ]]; then
          echo "Health Check Passed! Server is healthy."
        else
          echo "Health Check Failed! Server is not healthy."
          exit 1
        fi

```
Would something like this work?
@xoldyckk @palisadoes

**Comment by xoldyckk** [comment]
@aarishshahmohsin Take a look https://www.apollographql.com/docs/apollo-server/monitoring/health-checks/

**Comment by aarishshahmohsin** [comment]
```js
app.get("/", (req, res) =>
  res.json({
    "talawa-version": "v1",
    status: "healthy",
  })
);
```
This code is already there in the app.ts file. Is there any need of making a new endpoint?

**Comment by aarishshahmohsin** [comment]
> @aarishshahmohsin Take a look https://www.apollographql.com/docs/apollo-server/monitoring/health-checks/

```js
app.get("/", (req, res) =>
  res.json({
    "talawa-version": "v1",
    status: "healthy",
  })
);
```
This code is already there in the app.ts file. Is there any need of making a new endpoint?

**Comment by xoldyckk** [comment]
@aarishshahmohsin Since talawa-api is primarily a graphql api server the main thing to test is whether the apollo graphql engine is successfully responding to graphql based requests. 

I guess u should just test both things like mentioned in the docs I linked.

**Comment by aarishshahmohsin** [comment]
If there are any changes or information needed for the PR to be merged, please let me know. I'm eager to see this contribution integrated.

**Review by xoldyckk — COMMENTED** [review]
@aarishshahmohsin Instead of inline bash scripts, extract them out in files or better use typescript to write out the same tests. Just executing these tests should be handled in the github actions file.

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -123,4 +123,81 @@ jobs:
...
+        uses: actions/setup-node@v3
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Run dev server
+        run: npm run dev 
+
+      - name: Check health endpoint
+        run: bash ${GITHUB_WORKSPACE}/scripts/endpoint_health.sh
```
Comment (xoldyckk) [inline .github/workflows/pull-request.yml:200]: Does the **${GITHUB_WORKSPACE}** variable point to **/** or **/.github**?

Comment (aarishshahmohsin) [reply .github/workflows/pull-request.yml:200]: .github/

**Review by xoldyckk — APPROVED** [review]

**Comment by palisadoes** [comment]
@aarishshahmohsin Please fix the failing tests. The dev server workflow isn't working correctly

**Pull request closed without merging**